katharsis-client/src/main/java/io/katharsis/client/internal/ResourceRepositoryStubImpl.java

```diff
@@ -0,0 +1,141 @@
+		String requestBodyValue;
+		try {
+			requestBodyValue = objectMapper.writeValueAsString(context);
+		} catch (JsonProcessingException e) {
+			throw new IllegalStateException(e);
+		}
+
+		Builder builder = new Request.Builder().url(requestUrl);
+
+		builder = builder.post(RequestBody.create(null, requestBodyValue));
+
+		return execute(builder, true);
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public T findOne(ID id, QueryParams queryParams) {
+		HttpUrl url = urlBuilder.buildUrl(resourceClass, id, queryParams);
+		BaseResponseContext responseContext = executeGet(url);
+		return (T) responseContext.getResponse().getEntity();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<T> findAll(QueryParams queryParams) {
+		HttpUrl url = urlBuilder.buildUrl(resourceClass, null, queryParams);
+		CollectionResponseContext responseContext = (CollectionResponseContext) executeGet(url);
+		return (List<T>) responseContext.getResponse().getEntity();
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public List<T> findAll(Iterable<ID> ids, QueryParams queryParams) {
+		HttpUrl url = urlBuilder.buildUrl(resourceClass, ids, queryParams);
+		BaseResponseContext responseContext = executeGet(url);
+		if (responseContext instanceof CollectionResponseContext) {
+			CollectionResponseContext colResponseContext = (CollectionResponseContext) responseContext;
+			return (List<T>) colResponseContext.getResponse().getEntity();
+		} else {
+			return (List<T>) Arrays.asList(responseContext.getResponse().getEntity());
+		}
+	}
+
+	@Override
+	public <S extends T> S save(S entity) {
+		return save(entity, new QueryParams());
+	}
+
+	@Override
+	public <S extends T> S save(S entity, QueryParams queryParams) {
+		// String idFieldName =
+		// resourceInformation.getIdField().getUnderlyingName();
+		// send as post, no ids send
+		// consider using patch for udpates
+		Object id = null;
+		// try {
+		// id = PropertyUtils.getProperty(entity, idFieldName);
+		// } catch (Exception e) {
+		// throw new IllegalStateException(e);
+		// }
+
+		HttpUrl url = urlBuilder.buildUrl(resourceClass, id, null);
+		BaseResponseContext context = executePost(url, entity, queryParams);
+		return (S) context.getResponse().getEntity();
+	}
+
+	@Override
+	public void delete(ID id) {
+		HttpUrl url = urlBuilder.buildUrl(resourceClass, id, null);
+
+		executeDelete(url);
+	}
+
+	// // FIXME belongs to resource information
+	// @Deprecated
+	// public String getResourceType(Class clazz) {
+	// Class resourceClazz = ClassUtils.getJsonApiResourceClass(clazz);
+	// if (resourceClazz == null) {
+	// return null;
+	// }
+	// Annotation[] annotations = resourceClazz.getAnnotations();
+	// for (Annotation annotation : annotations) {
+	// if (annotation instanceof JsonApiResource) {
+	// JsonApiResource apiResource = (JsonApiResource) annotation;
+	// return apiResource.type();
+	// }
+	// }
+	// // won't reach this
+	// return null;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) This block of commented-out lines of code should be removed. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)